src/widget/wnumberrate.cpp

```diff
@@ -32,11 +31,14 @@ void WNumberRate::setup(const QDomNode& node, const SkinContext& context) {
 void WNumberRate::setValue(double dValue) {
     double vsign = dValue - 1;
 
-    char sign = '+';
-    if (vsign < -0.00000001) {
+    QString sign(' ');
+    if(vsign > 0) {
+        sign = '+';
+    }
+    if(vsign < 0) {
         sign = '-';
     }
 
-    setText(QString(m_skinText).append(sign)
+    setText(sign
```

Comment:
Where did m_skinText go?

Reply:
It was an empty variable anyway, I guess unused

Reply:
The m_skinText member is defined in the base class and might be populated elsewhere. Please don't modify this code unless you are able to justify the changes. A guess is not considered sufficient.

Reply:
But the string is empty and makes no sense. Maybe in the base class but not in WNumber. If you insist on having a variable having no value, I will revert this.

Reply:
`m_skinText` is a protected member of WLabel. It makes no sense to use it in this subclass.

Reply:
Just because it is empty doesn't mean that it is unused. It is customizable, though I am not aware if any skin actually provides a values. As you can see in the base class WNumber it **is** used.

Removing it here introduces even more inconsistency. If in doubt don't change it. The change is not required for the fix.

If this member is not supposed to be used then the widget hierarchy is wrong.

Reply:
The hierarchy is wrong now. I rolled back